test/integration/controller.v2/trainjob_controller_test.go

```diff
@@ -71,4 +73,120 @@ var _ = ginkgo.Describe("TrainJob controller", ginkgo.Ordered, func() {
 			gomega.Expect(k8sClient.Create(ctx, trainJob)).Should(gomega.Succeed())
 		})
 	})
+
+	ginkgo.When("TrainJob CR Validation", func() {
+		ginkgo.AfterEach(func() {
+			gomega.Expect(k8sClient.DeleteAllOf(ctx, &kubeflowv2.TrainJob{}, client.InNamespace(ns.Name))).Should(
+				gomega.Succeed())
+		})
+
+		ginkgo.It("Should succeed in creating TrainJob", func() {
+
+			managedBy := "kubeflow.org/trainjob-controller"
+
+			trainingRuntimeRef := kubeflowv2.RuntimeRef{
+				Name:     "InvalidRuntimeRef",
+				APIGroup: &apiGroup,
+				Kind:     &trainingRuntimeKind,
```

Comment:
```suggestion
				APIGroup: ptr.To(kubeflowv2.GroupVersion.Group),
				Kind:     ptr.To(trainingRuntimeKind),
```

Could you follow the same approach in other places?